qt/aqt/browser.py

```diff
@@ -786,155 +381,70 @@ def add_preview_button(leftbuttons: List[str], editor: Editor) -> None:
         self.editor = aqt.editor.Editor(self.mw, self.form.fieldsArea, self)
         gui_hooks.editor_did_init_left_buttons.remove(add_preview_button)
 
-    def onRowChanged(self, current: QItemSelection, previous: QItemSelection) -> None:
+    def save_now(self, callback: Callable) -> None:
+        self.editor.saveNow(callback)
+
+    def begin_reset(self) -> None:
+        """Must be called before modifying the current note from outside the editor to prevent
+        overwriting the changes with the cache. Unsaved changes in the editor will be lost.
+        After the modification, _onRowChanged() must be triggered, e.g. by end_reset() or search(),
+        to move the editor to a consistent state.
+        """
+        self.editor.setNote(None, hide=False)
+
```

Comment:
This is different to the old beginReset() and endReset() behaviour, and I'm not 100% sure it's safe at the moment. If changes are made such as deleting a note without beginReset() being called on the model, Qt may queue up a request that tries to redraw a row that references that note, and if that request is processed prior to the model being reset, we could end up getting an error. This was a big issue in the past when the DB layer would periodically call back into Python and drain the UI event queue, but I'm not sure how much of an issue it is now.

Reply:
You are right, I realised that `delete_notes` could be a potential problem after changing the resetting process. Maybe that method could be handled separately as there isn't much else that could go wrong.
I'm not sure if it's possible to trigger that error at all, though, as the notes should already be loaded into the cache.